hack/verify-deps.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -euo pipefail
+
+function print_failure {
+  echo "There are unexpected changes to the vendor tree following 'go mod vendor' and 'go mod tidy'. Please"
+  echo "run these commands locally and double-check the Git repository for unexpected changes which may"
+  echo "need to be committed."
+  exit 1
+}
+
+if [ "${OPENSHIFT_CI:-false}" = true ]; then
+  go mod vendor
```

Comment:
These ones change some file on `main`:
```
$ git st --porcelain
$ go mod vendor && go mod tidy
$ git st --porcelain
 M go.mod
 M vendor/modules.txt
```
I think we can add the changes to this PR to start clean.

Reply:
We could start clean by using some temporary commit logic? Thoughts?

Reply:
For reference, the OPENSHIFT_CI flag is what stops this script from interfering with any locally uncommitted changes (at the cost of not actually running when you run `make verify`). The expectation in CI is that all changes will be committed (since CI will do a git clone).

Reply:
I was just thinking about the next guy who'll do a PR, it would fail due to `verify-deps.sh`.
But it won't as there is no CI job for it but once it's there it will.

Well, it's not a big deal anyway; LGTM.

Reply:
Oh, I misunderstood. Ill open a new PR to fix that :grin: